pkgs/by-name/qh/qhexedit2/package.nix

```diff
@@ -0,0 +1,76 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  qt6,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "qhexedit2";
+  version = "0.8.9";
+
+  src = fetchFromGitHub {
+    owner = "Simsys";
+    repo = "qhexedit2";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-qg8dyXwAsTVSx85Ad7UYhr4d1aTRG9QbvC0uyOMcY8g=";
+  };
+
+  postPatch = ''
+    # Replace QPallete::Background with QPallete::Window in all files
+    find . -type f -exec sed -i 's/QPalette::Background/QPalette::Window/g' {} +
+  '';
+
+  nativeBuildInputs = [
+    qt6.qmake
+    qt6.wrapQtAppsHook
+  ];
+  buildInputs = [
+    qt6.qtbase
+    qt6.qttools
+  ];
+
+  qmakeFlags = [
+    "./example/qhexedit.pro"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/lib $out/include
+
+    # Copy all .o files to $out/lib
+    for file in $(find . -name '*.o'); do
+      cp $file $out/lib
+    done
+
+    # Copy all .so files to $out/lib
+    for file in $(find . -name '*.so'); do
+      cp $file $out/lib
+    done
+
+    # Copy all .cpp files to $out/include
+    for file in $(find . -name '*.cpp'); do
+      cp $file $out/include
+    done
```

Comment:
According to `nix-locate -r '/include.*\.cpp$'`, copying `.cpp` files seems to be common, and since `qhexedit2` can be used as a library, arguably they should be copied. I guess the `.o` files don't need to be copied if the `.so` files are.

Reply:
That's a good point, and I can't remember either. I'll remove that too. I wonder why the other packages do it.